src/realm/sync/subscriptions.cpp

```diff
@@ -277,7 +272,7 @@ std::pair<SubscriptionSet::iterator, bool> MutableSubscriptionSet::insert_or_ass
     auto table_name = Group::table_name_to_class_name(query.get_table()->get_name());
     auto query_str = query.get_description();
     auto it = std::find_if(begin(), end(), [&](const Subscription& sub) {
-        return (sub.has_name() && sub.name() == name);
+        return (sub.name() && *sub.name() == name);
```

Comment:
```suggestion
        return (sub.name() == name);
```

`==` does the right thing with `Optional<T>`